app/controllers/casa_cases_controller.rb

```diff
@@ -60,7 +60,9 @@ def update
 
   def deactivate
     casa_case = CasaCase.find(params[:id])
-    # TODO: authorize action with pundit
+
+    return render :edit unless Pundit.policy(current_user, casa_case).update_case_status?
```

Comment:
**Qquestion:** Do we want to return any specific message in the `notice`? We could return some message that explains that there was an error but does not tell the user about the missing permission (kinda like security by obscurity). 
For now, I'm just rendering the `edit` form in case of unauthorized.

Reply:
If they aren’t authorized to edit, I would maybe redirect them back to where they came from. Index/dashboard?

Reply:
Also, I’m a fan of adding a flash message. But @Zrrrpy  would probably have a better understanding of what the users expect.

Reply:
> If they aren’t authorized to edit, I would maybe redirect them back to where they came from. Index/dashboard?

Yeah, I believe redirect to referrer would be a good way forward. I'm also a fan of flash messages, but there were situations in the past where the flash was not persisted with redirects. I might be mixing up memories though haha 😅 .

I've also realised that these actions in specific do not follow the pattern of the other ones in this controller where we  `respond_to` both `html` and `json` formats.

Reply:
This would be the correct way to handle the authorization in the controller, with the `authorize` check:

```
def deactivate
  @casa_case = CasaCase.find(params[:id])
  authorize @casa_case, :update_case_status?
  flash[:notice] = "blah"
  render :edit
end
```

And then you don't need to explicitly decide what to do with an unauthorized user since that's taken care of [automatically in the application controller](https://github.com/rubyforgood/casa/blob/main/app/controllers/application_controller.rb#L8).

Reply:
And I wouldn't worry about responding to the `json` format, we don't have any API set up currently.

Reply:
Hey thanks, those are very good points, I just updated the PR now.
I've added specs for these methods, there was none before. The specs already helped me to find some issues and I got them fixed. 🎉